File: app/src/main/java/com/github/haocen2004/login_simulation/Database/Sponsor/SponsorDatabase.java
package com.github.haocen2004.login_simulation.Database.Sponsor;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {SponsorData.class},version = 1,exportSchema = false)
public abstract class SponsorDatabase extends RoomDatabase {
    private static SponsorDatabase INSTANCE;
    static synchronized SponsorDatabase getDatabase(Context context){
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(context.getApplicationContext(), SponsorDatabase.class, "sponsors_db")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return INSTANCE;
    }
    public abstract SponsorDao getSponsorDao();
}

File: app/src/main/java/com/github/haocen2004/login_simulation/data/sponsor/database/SponsorDatabase.kt
package com.github.haocen2004.login_simulation.data.sponsor.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase

@Database(entities = [SponsorData::class], version = 1, exportSchema = false)
abstract class SponsorDatabase : RoomDatabase() {
    abstract fun sponsorDao(): SponsorDao

    companion object {
        private var INSTANCE: SponsorDatabase? = null

        fun getInstance(context: Context): SponsorDatabase {
            return INSTANCE ?: synchronized(this) {
                val instance = databaseBuilder(
                    context,
                    SponsorDatabase::class.java,
                    "sponsors_db"
                ).build()

                INSTANCE = instance
                instance
            }
        }
    }
}